FitGoal/FitGoal/Singup&Singin/UserInformationPersistenceManager.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  UserInformationPersistanceManager.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 1/4/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import Foundation
+struct UserInformationPersistenceManager {
+    
+    var userInfo: UserInformation?
+    
+    init(userInfo: UserInformation) {
+        self.userInfo = userInfo
+    }
+    
+    func saveUserInformation() {
+        guard let userInfo = self.userInfo else { return }
+        let defaults = UserDefaults.standard
+        defaults.set(userInfo.name, forKey: UsersInfoKey.name.rawValue)
+        defaults.set(userInfo.email, forKey: UsersInfoKey.email.rawValue)
+    }
+    
+    static func isUserLogedIn() -> Bool {
```

Comment:
The interface of this class is problematic. Consider the following problems:
- You can't ask to an instance of the struct if there is a logged in user.
- You expose the `userInfo` field as an optional type, however, it won't ever be optional.
- Creating an instance of the struct with a userInfo in the initialiser means nothing, since you don't save the data there. After creating the struct you are required to call the `saveUserInformation` function right away, which is a source of bugs. Why not call the function from the initialiser?
- Try to always have functions in pairs (when it makes sense). For example, if you have an `add` function in a class, you probably need a `remove` too. In this case, if you have a `save`, you probably need a `getInfo` too.  With the current implementation you can neither show the information of the logged-in user in the app, nor logout the user by clearing out the information.

A better approach were all these problems are fixed would be something like:

```swift
class AppPreferences {
    var loggedInUser: UserInformation? {
        get { // get value from UserDefaults }
        set { // set value to UserDefaults }
    }
}
```

Reply:
Yeah! This is better, thanks